src/ec/packages/ec-component-footer-harmonised/footer-harmonised.story.js

```diff
@@ -35,18 +31,399 @@ groups.forEach(g => {
+  dataG1 = [...dataGroup1.sections];
+  dataG1[1][0] = contactUs;
+  dataG1[1][1] = followUs;
+  dataG1[2][0] = aboutUs;
+  dataG1[2][1] = related;
+};
+
+// Prepare the knobs for group3.
+const formatFooterG3 = dataG3 => {
+  dataG3[0].title = text('sections[0].title', dataG3[0].title, requiredGroupId);
+  dataG3[1].logos.forEach((logo, index) => {
+    let label = optionalGroupId;
+    let logoSrc = dataG3[1].logos[index].logo.src;
+    if (index === 2) {
+      label = requiredGroupId;
+      logoSrc = logoEC;
+    }
+    if (logo) {
+      dataG3[1].logos[index].logo.title = text(
+        `dataG3[1].logos[${index}].logo.title`,
+        dataG3[1].logos[index].logo.title,
+        label
+      );
+      dataG3[1].logos[index].logo.alt = text(
+        `dataG3[1].logos[${index}].logo.alt`,
+        dataG3[1].logos[index].logo.alt,
+        label
+      );
+      dataG3[1].logos[index].logo.src = text(
+        `dataG3[1].logos[${index}].logo.src`,
+        logoSrc,
+        label
+      );
+    }
+  });
+
+  // Return the full specs.
+  return dataG3;
+};
+
+// Prepare the knobs for group2.
+const formatFooterG2 = dataG2 => {
+  // Corporate name.
+  dataG2[0].title.link = {
+    label: text(
+      'sections[0].title.link.label',
+      dataG2[0].title.link.label,
+      requiredGroupId
+    ),
+    path: text(
+      'sections[0].title.link.path',
+      dataG2[0].title.link.path,
+      requiredGroupId
+    ),
+    placeholder: text('', 'none', optionalGroupId),
+  };
+  // Service navigation.
+  dataG2[1].links.forEach((link, index) => {
+    dataG2[1].links[index].link.label = text(
+      `sections[1].links[${index}].link.label`,
+      dataG2[1].links[index].link.label,
+      requiredGroupId
+    );
+    dataG2[1].links[index].link.path = text(
+      `sections[1].links[${index}].link.path`,
+      dataG2[1].links[index].link.path,
+      requiredGroupId
+    );
+  });
+  // Legal navigation.
+  dataG2[2].links.forEach((link, index) => {
+    dataG2[2].links[index].link.label = text(
+      `sections[2].links[${index}].link.label`,
+      dataG2[2].links[index].link.label,
+      requiredGroupId
+    );
+    dataG2[2].links[index].link.path = text(
+      `sections[2].links[${index}].link.path`,
+      dataG2[2].links[index].link.path,
+      requiredGroupId
+    );
+  });
+
+  // Return the full specs.
+  return dataG2;
+};
+
+// Prepare the knobs for group1
+const formatFooterG1 = dataG1 => {
+  // Swap the columns when needed.
+  if (!dataG1[1][0].title && !dataG1[1][1].title) {
+    dataG1[1][0] = dataG1[2][0].title ? aboutUs : { section_id: 2 };
+    dataG1[1][1] = dataG1[2][1].title ? related : { section_id: 2 };
+    dataG1[1][0].section_id = 2;
+    dataG1[1][1].section_id = 2;
+    dataG1[2] = [{ section_id: 3 }, { section_id: 3 }];
+  }
+  // Site name & content owner details.
+  dataG1[0].title.link = {
+    label: text(
+      'sections[0].title.link.label',
+      dataG1[0].title.link.label,
+      requiredGroupId
+    ),
```

Comment:
I think here also a better name for the sections and a beter order in the sections should help here. You can always look the find the knob by its value but still would it be nice if this was in order.

Reply:
I appreciate the fact that you have in mind something better than this naming, but why don't you tell me? Saying "a better name would help" means that you have an idea on what it could be, generally speaking what i aim for is that we define the labels for knobs using the real property in the real "object" we send as an input data. In this case i see that i haven't been consistent in the three footers, but the example you put here is the "right" one, 
`sections[0].title.link.label`
is the real way to access that property in the input data, indeed while sometimes i wrongly used the name of the variable used in the story, i will change this.

Reply:
I don't think we need to change it completely but for instance we could call it `Section 0 link label` and if we have multiple links in the section we could call it `Section 0 link 1 label` and `Section 0 link 1 path`. To me this would look nicer but if you still want to go with sections[0].title.link.label its fine by me as well